### ISSUE TITLE
At install time, allow user to install and enable official Cacti plugins

### ISSUE THREAD

**Issue by cigamit** [Cacti/cacti#1376]
Right now, all plugins must be downloaded and installed manually.  It would be nice it at install time, or via some other interface, a user could select a list of official plugins to download and install automatically.

**Comment by netniV** [comment]
We'd have to check for write permissions to the plugins directory.

**Comment by cigamit** [comment]
Correct.  Right now, at install time, we need write permissions for a few pages.  We warn the user to undo that after install. I have seen various methods of providing a more secure interface to write files by creating another web service that takes trusted actions on the part of the web server.  This might be a better long term solution.

**Comment by netniV** [comment]
Yes it would, because it would also allow the installation of Script Server code too.  It's a pain at the moment having to manually copy that into the right place.

**Comment by netniV** [comment]
You should have opened a separate issue for your issue.  Please do so if you have any further questions about, this is a different thread.  For your information, mURLin has not been updated for 1.x.   I was waiting on someone who has done the work to upload it to a repo I created but so far that hasn't happened yet so keep an eye on https://github.com/netniV/plugin_murlin

**Comment by jpobeda** [comment]
Make sure to warn the user that it'll need internet access and allow the user to specify a proxy server.

**Comment by cigamit** [comment]
Agree on this, but I'm going to disagree on the approach just a little.  We have to do one of the following:

- Create a repository somewhere that allows people to perform pull requests to include their plugins in automatic download, a manifest file, and copies of plugins, not only from the Cacti group, but others that are included.

or

- A repo management system like Yellowdog that allows publishers to publish a location where user can obtain a list of plugins and use the method above in each.

or

- Each plugin define an update location where the user can pull the manifest as articulated by @netniV in his initial post.

I prefer 1) or 2) as it allows people to publish them to the Cacti group and provides a central depot for users around the world, but would be Okay, with 2) as well, cause RedHat and others are too.  I'm not as happy with 3) as it does not really encourage sharing and community.

**Comment by cigamit** [comment]
This would also support upgrades as I have discussed in another issue/thread.

**Comment by bmfmancini** [comment]
@TheWitness  I think this can be closed now with the new changes

**Comment by netniV** [comment]
Aren't the new changes related to templates, not plugins?

**Comment by TheWitness** [comment]
Correct.  This has not been delivered as yet.

**Comment by TheWitness** [comment]
This is now delivered and kicked I 1.3! Merry Christmas
